db/migrate/20180806151925_add_subnet_name_unique_constraint.rb

```diff
@@ -0,0 +1,27 @@
+class AddSubnetNameUniqueConstraint < ActiveRecord::Migration[5.1]
+  def up
+    merge_duplicate_subnets
+    # we need to specify the lenght to make MySQL happy
+    add_index :subnets, :name, :unique => true, length: { name: 255 }
+  end
+
+  def down
+    remove_index :subnets, :name
+  end
+
+  private
+
+  def merge_duplicate_subnets
+    duplicate_names = Subnet.unscoped.select(:name).group(:name).having('count(name) > 1').pluck(:name)
+    Rails.logger.info("Found #{duplicate_names} duplicate subnet names.")
+    duplicate_names.each do |name|
+      winner, *loosers = Subnet.unscoped.where(name: name)
```

Comment:
How about just `#pluck`ing the ids here and doing something like `Subnet.unscoped.where(id: loosers).destroy_all` on the last line instead of retrieving the ids from the `Subnet` objects all the time?

Reply:
I went for `destroy` just in case there was any cleanup happening in the after_destroy hooks. On the other hand, I can imagine this causing troubles in case the after_destoy hooks used some attributes that were not migrated yet. I will update that later today